test/provider_os_memory_multiple_numa_nodes.cpp

```diff
@@ -101,9 +149,342 @@ TEST_P(testNumaNodes, checkNumaNodesAllocations) {
+// triggered the allocation. It will be executed on each available CPU.
+TEST_P(testNumaOnEachCpu, checkModePreferredEmptyNodeset) {
+    int cpu = GetParam();
+    ASSERT_GE(cpu, 0);
+
+    cpu_set_t *mask = CPU_ALLOC(CPU_SETSIZE);
+    CPU_ZERO(mask);
+
+    CPU_SET(cpu, mask);
+    int ret = sched_setaffinity(0, sizeof(cpu_set_t), mask);
+    UT_ASSERTeq(ret, 0);
+
+    umf_os_memory_provider_params_t os_memory_provider_params =
+        UMF_OS_MEMORY_PROVIDER_PARAMS_TEST;
+    os_memory_provider_params.numa_mode = UMF_NUMA_MODE_PREFERRED;
+    initOsProvider(os_memory_provider_params);
+
+    umf_result_t umf_result;
+    umf_result =
+        umfMemoryProviderAlloc(os_memory_provider, alloc_size, 0, &ptr);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    ASSERT_NE(ptr, nullptr);
+
+    // Verify we're on the expected CPU
+    int cpu_check = sched_getcpu();
+    UT_ASSERTeq(cpu, cpu_check);
+
+    int numa_node_number = numa_node_of_cpu(cpu);
+    printf("Got CPU: %d, got numa node: %d\n", cpu, numa_node_number);
+
+    // 'ptr' must point to an initialized value before retrieving its numa node
     memset(ptr, 0xFF, alloc_size);
     int retrieved_numa_node_number = getNumaNodeByPtr(ptr);
     ASSERT_EQ(retrieved_numa_node_number, numa_node_number);
+    CPU_FREE(mask);
+}
+
+// Test for allocation on numa node with local mode enabled. The memory is
+// allocated on the node of the CPU that triggered the allocation.
+// It will be executed on each available CPU.
+TEST_P(testNumaOnEachCpu, checkModeLocal) {
+    int cpu = GetParam();
+    cpu_set_t *mask = CPU_ALLOC(CPU_SETSIZE);
+    CPU_ZERO(mask);
+
+    CPU_SET(cpu, mask);
+    int ret = sched_setaffinity(0, sizeof(cpu_set_t), mask);
+    UT_ASSERTeq(ret, 0);
+
+    umf_os_memory_provider_params_t os_memory_provider_params =
+        UMF_OS_MEMORY_PROVIDER_PARAMS_TEST;
+    os_memory_provider_params.numa_mode = UMF_NUMA_MODE_LOCAL;
+    initOsProvider(os_memory_provider_params);
+
+    umf_result_t umf_result;
+    umf_result =
+        umfMemoryProviderAlloc(os_memory_provider, alloc_size, 0, &ptr);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    ASSERT_NE(ptr, nullptr);
+
+    // Verify we're on the expected CPU
+    int cpu_check = sched_getcpu();
+    UT_ASSERTeq(cpu, cpu_check);
+
+    int numa_node_number = numa_node_of_cpu(cpu);
+    printf("Got CPU: %d, got numa node: %d\n", cpu, numa_node_number);
+
+    // 'ptr' must point to an initialized value before retrieving its numa node
+    memset(ptr, 0xFF, alloc_size);
+    int retrieved_numa_node_number = getNumaNodeByPtr(ptr);
+    ASSERT_EQ(retrieved_numa_node_number, numa_node_number);
+    CPU_FREE(mask);
+}
+
+// Test for allocation on numa node with default mode enabled.
+// Since no policy is set by the set_mempolicy function, it should
+// default to the system-wide default policy, which allocates pages
+// on the node of the CPU that triggers the allocation.
+TEST_F(testNuma, checkModeDefault) {
```

Comment:
This test could be also ran on all possible cpus.

Reply:
Done.

Reply:
hmm... or not done... it started to failing with this change. I reverted it and will debug it later.